### PR TITLE
DOC-10371 Add note regarding supported interpolation methods for Series/DF

### LINKED ISSUE #10371: Add note in docs for interpolate that Only `method=linear` interpolation is supported on MultiIndexes.
Trying to interpolate a DataFrame with MultiIndex:
```
lib/python2.7/site-packages/pandas/core/generic.pyc in interpolate(self, method, axis, limit, inplace, downcast, **kwargs)
   2868 
   2869         if isinstance(self.index, MultiIndex) and method != 'linear':
-> 2870             raise ValueError("Only `method=linear` interpolation is supported "
   2871                              "on MultiIndexes.")
   2872 

ValueError: Only `method=linear` interpolation is supported on MultiIndexes.
```
add a note [in the docs](http://pandas.pydata.org/pandas-docs/dev/generated/pandas.Series.interpolate.html).

### PR DESCRIPTION
**Pull request by Winterflower**
Closes #10371

Suggested improvement for https://github.com/pydata/pandas/issues/10371

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -2875,6 +2875,8 @@ def interpolate(self, method='linear', axis=0, limit=None, inplace=False,
                     downcast=None, **kwargs):
         """
         Interpolate values according to different methods.
+        Please note that only method='linear' is supported for DataFrames/Series
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: Can you leave a blank line between the first sentence and this new one?

**Comment by jorisvandenbossche** [comment]
Looks good! 
About the formatting, for me this is good as well, or otherwise make it look like code as: ``` ``method='linear'`` ```

**Comment by Winterflower** [comment]
Thanks @jorisvandenbossche!
Incorporated your suggestions.
Local build of the docstring looks ok.

**Comment by jorisvandenbossche** [comment]
Yep, looks good. Can you just squash the two commits into one?

**Comment by Winterflower** [comment]
Sounds like a recipe for one of my famous git disasters, but I'll give it a go.

**Comment by jorisvandenbossche** [comment]
it is totally not difficult :-), just ``git rebase -i master`` and then force push

**Comment by jorisvandenbossche** [comment]
Merging now, thanks a lot!

**Comment by Winterflower** [comment]
@jorisvandenbossche did the squashing really work?  Wow, that must be the first time I do something like that in git without borking a million other files at the same time! Thanks!

**Comment by jorisvandenbossche** [comment]
I don't know *what* exactly you did, but in any case, it was OK! 
BTW, really nice to see contribution to the docs! Hopefully there will follow some more :-)

**Pull request merged**